web/themes/joinup/joinup_theme.theme

```diff
@@ -1092,21 +1092,19 @@ function joinup_theme_preprocess_comment(&$variables) {
     '#account' => $user,
   ];
 
+  $cache_metadata = (new CacheableMetadata())->addCacheableDependency($user);
+  $cache_metadata->applyTo($variables['author']);
+
   if ($photo_target) {
     $photo_target = $photo_target[0]['target_id'];
     /** @var \Drupal\file\Entity\File $file */
     if ($file = File::load($photo_target)) {
       $uri = $file->getFileUri();
       $url = ImageStyle::load('profile_icon_big')->buildUrl($uri);
-      $variables['profile_icon'] = [
-        '#markup' => $url,
-        '#cache' => [
-          'tags' => Cache::mergeTags($user->getCacheTags(), $user->getEntityType()
-            ->getListCacheTags()),
-          'contexts' => $user->getCacheContexts(),
-          'max-age' => $user->getCacheMaxAge(),
-        ],
-      ];
+      $variables['profile_icon'] = ['#markup' => $url];
+      $cache_metadata
+        ->addCacheTags($user->getEntityType()->getListCacheTags())
```

Comment:
As far as I understand you are trying to make the user icon in the comment cache-able but you are only passing the entity type list cache tags. But is this one even needed? The icon should not be affected if _any_ account changes, just the current one, so this should carry the `$user->getCacheTags()` instead. 
I am not sure if the user specific cache tags are invalidated when the user is deleted but I can assume they are.
I see for user 1 for example,
```
>>> $user->getCacheTags();
=> [
     "user:1",
   ]
>>> $user->getEntityType()->getListCacheTags();
=> [
     "user_list",
   ]
>>>
```
 So normally, if the `$user->getCacheTags()` are invalidated during account deletion then these should be enough. But actually, now with the new cancellation-no-deletion method, we don't even need to worry about the deletion.